### PR TITLE
Graph transform

### PR DESCRIPTION
**Pull request by shenoynikhil**
## Proposed changes

Fixes for issue #123. 

#### Description
Adds an ability to apply a callable transform to graphdata during `__getitem__`.

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

**Comment by TristanBilot** [comment]
Hi @shenoynikhil, thanks for the PR! Would you mind adding some tests?

**Comment by shenoynikhil** [comment]
Sorry for the delay. I added the test. It took me a bit to setup. I think the latest code in there requires `mlx==0.6.*`.

File: tests/datasets/test_dataset.py [hunk 1]
```diff
@@ -80,3 +80,32 @@ def process(self):
...
+        def download(self):
+            pass
+
+        def process(self):
+            self.graphs = data_list
+
+    dataset = Dataset1(custom_transform)
+
+    for i in range(len(dataset)):
+        assert dataset[i].feat == "test_transform"
```
Comment (francescofarina) [inline tests/datasets/test_dataset.py:111]: Can you delete the content of `path` after this?
We'll have to implement a fixture to automatically do that for all files created during testing now that we have multiple datasets, but we'll do that in a later PR.

**Review by francescofarina — CHANGES_REQUESTED** [review]

File: tests/datasets/test_dataset.py (hunk 1 above)
Comment (shenoynikhil) [reply tests/datasets/test_dataset.py:111]: Added a line to remove the files at the path.

**Review by francescofarina — APPROVED** [review]
Looks good! I did some minor tweaking to the docs but it's great.

We may want to implement a `pre_transform` as well in a following PR to pre-transform all data before saving them @shenoynikhil, in case you're interested in looking into that.

**Pull request merged**